include/boost/url/experimental/detail/impl/url_template.ipp

```diff
@@ -0,0 +1,379 @@
+            }
+            if (has_pass)
+            {
+                if (pass.starts_with('{'))
+                {
+                    n.pass = visit_format_arg(
+                        make_measure_visitor(
+                            urls::detail::password_chars),
+                        args.get(idx++));
+                }
+                else
+                {
+                    n.pass = pass.size();
+                }
+            }
+        }
+        if (host.starts_with('{'))
+        {
+            n.host = visit_format_arg(
+                make_measure_visitor(
+                    urls::detail::host_chars),
+                args.get(idx++));
+        }
+        else
+        {
+            n.host = host.size();
+        }
+        if (has_port)
+        {
+            if (port.starts_with('{'))
+            {
+                n.port = visit_format_arg(
+                    make_measure_visitor(
+                        grammar::lut_chars(
+                            "0123456789")),
+                    args.get(idx++));
+            }
+            else
+            {
+                n.port = port.size();
+            }
+        }
+    }
+    auto it = path.data();
+    auto end = path.data() + path.size();
+    while (it != end)
+    {
+        if (*it == '/')
+        {
+            ++n.path;
+            ++it;
+            continue;
+        }
+        if (*it == '{')
+        {
+            while (
+                it != end &&
+                *it != '}')
+                ++it;
+            BOOST_ASSERT(it != end);
+            ++it;
+            n.path += visit_format_arg(
+                make_measure_visitor(
+                    urls::detail::path_chars),
+                args.get(idx++));
+            continue;
+        }
+        while (
+            it != end &&
+            *it != '/')
+        {
+            ++n.path;
+            ++it;
+        }
+    }
+    if (query.starts_with('{'))
+    {
+        n.query = visit_format_arg(
+            make_measure_visitor(
+                urls::detail::query_chars),
+            args.get(idx++));
+    }
+    else
+    {
+        n.query = query.size();
+    }
+    if (frag.starts_with('{'))
+    {
+        n.frag = visit_format_arg(
+            make_measure_visitor(
+                urls::detail::fragment_chars),
+            args.get(idx++));
+    }
+    else
+    {
+        n.frag = frag.size();
+    }
+    u.reserve(
+        n.scheme +
+        (n.scheme != 0) * 1 +
+        has_authority * 2 +
+        n.user +
+        has_pass * 1 +
+        n.pass +
+        has_user * 1 +
+        n.host +
+        has_port * 1 +
+        n.port +
+        n.path +
+        has_query * 1 +
+        n.query +
+        has_frag * 1 +
+        n.frag);
+
+    // Apply
+    idx = 0;
+    using parts = urls::detail::parts_base;
+    if (!scheme.empty())
+    {
+        if (scheme.starts_with('{'))
+        {
+            url_base::op_t op(u);
```

Comment:
If create and destroy an instance of `opt_t` more than once in a function, then you will get the Basic exception guarantee and not the Strong.

Reply:
I'll do that anyway but a single `opt_t` won't get us a strong guarantee either, right?

Reply:
yes it does